docs/adr/012_decouple_from_edge.md

```diff
@@ -0,0 +1,147 @@
+
+When a leaf device message is sent and processed, the device twin gets updated with the "GatewayID", currently being set to the "IOTEDGE_DEVICEID" environment variable.
+
+This information might be used by the Azure Function:
+
+- for sending the LNS a command to close the device client connection for such leaf device
+- for sending a C2D message for a Class-C device through a "direct method invocation"
+
+We need to **conditionally set the "GatewayID" parameter depending on whether the LNS is running on Edge or not**
+
+The proposal is to:
+
+- Prefix the "GatewayID" with "standalone-" for non-Edge deployments
+- Use the "Hostname" environment variable as a default for GatewayID field
+
+### Managing configuration and its changes
+
+All the configuration is fetched at the start time from the environment variables.
+
+IoT Edge allows you to specify those configuration variables as part of a template for [automatic deployments for single devices or at scale](https://docs.microsoft.com/en-us/azure/iot-edge/module-deployment-monitoring?view=iotedge-2020-11)
+
+Depending on the target cloud service, you might have different ways to manage configuration in a centralized way.
+
+If using Azure Kubernetes Service, a configmap to be applied to the desired cluster might work for you.
+If using Azure App Service, you might just set the environment variables in the configuration section of the application.
+
+While these solutions are great for "static" configuration (i.e.: those values that require a restart for being changed), the LNS is coded in such a way that some configuration might be changed dynamically without having to restart the entire network server.
+
+The proposal is to **make use of "[Azure App Configuration](https://docs.microsoft.com/en-us/azure/azure-app-configuration/overview)"** to centrally manage application settings and keep [dynamic configuration using poll model](https://docs.microsoft.com/en-us/azure/azure-app-configuration/enable-dynamic-configuration-aspnet-core?tabs=core5x)
+
+### Direct method invocation
+
+As previously mentioned, IoT Edge is allowing the functionality of "direct method invocations" giving the possiblity to remotely invoke some functions on the LNS.
+
+When decoupling LNS from Azure IoT Edge, there are two possibilities available for handling the scenarios covered by direct method invocation.
+
+For both mechanisms the "Facade" Azure Function bundled with the Starter Kit, has to differentiate between a "edge" or "cloud" deployment of LNS. The proposal is to **differentiate the mechanism used for remotely invoke methods depending on the "LNS device id"**. This means that an Azure IoT Edge device will still be using the direct method invocations provided by IoT Hub whereas the "cloud" deployments will use a new, different, mechanism.
+
+#### Expose REST endpoints on the LNS
```

Comment:
this is repeating the problem from above. I would suggest to create a dedicated section, that talks about how to address the LNS in the 2 different deployment models and then link to that.